From deed4dc5fc2e5726d75fe538f1a71f1c7c2f1c30 Mon Sep 17 00:00:00 2001
From: Yuhang Zhao <2546789017@qq.com>
Date: Sun, 19 Jun 2022 13:06:56 +0800
Subject: [PATCH] wangwenx190: Custom tweaks

Import changes list:

01. MSVC: Don't embed debug symbols into binary file.
02. MSVC: Enable more optimizations.
03. All: disable debug builds.
04. FFmpeg: Don't enable debug configuration for release builds.

Signed-off-by: Yuhang Zhao <2546789017@qq.com>
---
 ports/ffmpeg/portfile.cmake                      |  2 +-
 scripts/toolchains/windows.cmake                 | 16 ++++++++--------
 triplets/arm-uwp.cmake                           |  1 +
 triplets/arm64-windows.cmake                     |  1 +
 triplets/community/arm-android.cmake             |  2 ++
 triplets/community/arm-ios.cmake                 |  2 ++
 triplets/community/arm-linux.cmake               |  1 +
 triplets/community/arm-mingw-dynamic.cmake       |  3 ++-
 triplets/community/arm-mingw-static.cmake        |  1 +
 triplets/community/arm-neon-android.cmake        |  2 ++
 triplets/community/arm-windows-static.cmake      |  1 +
 triplets/community/arm-windows.cmake             |  1 +
 triplets/community/arm64-android.cmake           |  2 ++
 triplets/community/arm64-ios.cmake               |  2 ++
 triplets/community/arm64-linux.cmake             |  1 +
 triplets/community/arm64-mingw-dynamic.cmake     |  1 +
 triplets/community/arm64-mingw-static.cmake      |  1 +
 triplets/community/arm64-osx-dynamic.cmake       |  1 +
 triplets/community/arm64-osx.cmake               |  1 +
 triplets/community/arm64-uwp.cmake               |  1 +
 triplets/community/arm64-windows-static-md.cmake |  1 +
 triplets/community/arm64-windows-static.cmake    |  2 +-
 triplets/community/armv6-android.cmake           |  2 ++
 triplets/community/ppc64le-linux.cmake           |  1 +
 triplets/community/s390x-linux.cmake             |  1 +
 triplets/community/wasm32-emscripten.cmake       |  2 ++
 triplets/community/x64-android.cmake             |  2 ++
 triplets/community/x64-freebsd.cmake             |  1 +
 triplets/community/x64-ios.cmake                 |  2 ++
 triplets/community/x64-linux-dynamic.cmake       |  1 +
 triplets/community/x64-linux-release.cmake       |  6 ------
 triplets/community/x64-mingw-dynamic.cmake       |  3 ++-
 triplets/community/x64-mingw-static.cmake        |  1 +
 triplets/community/x64-openbsd.cmake             |  1 +
 triplets/community/x64-osx-dynamic.cmake         |  1 +
 triplets/community/x64-osx-release.cmake         |  7 -------
 triplets/community/x64-windows-release.cmake     |  4 ----
 triplets/community/x64-windows-static-md.cmake   |  1 +
 triplets/community/x86-android.cmake             |  2 ++
 triplets/community/x86-freebsd.cmake             |  1 +
 triplets/community/x86-ios.cmake                 |  2 ++
 triplets/community/x86-mingw-dynamic.cmake       |  3 ++-
 triplets/community/x86-mingw-static.cmake        |  1 +
 triplets/community/x86-uwp.cmake                 |  1 +
 triplets/community/x86-windows-static-md.cmake   |  1 +
 triplets/community/x86-windows-static.cmake      |  3 ++-
 triplets/community/x86-windows-v120.cmake        |  2 ++
 triplets/x64-linux.cmake                         |  1 +
 triplets/x64-osx.cmake                           |  1 +
 triplets/x64-uwp.cmake                           |  1 +
 triplets/x64-windows-static.cmake                |  2 +-
 triplets/x64-windows.cmake                       |  2 +-
 triplets/x86-windows.cmake                       |  2 +-
 53 files changed, 73 insertions(+), 34 deletions(-)
 delete mode 100644 triplets/community/x64-linux-release.cmake
 delete mode 100644 triplets/community/x64-osx-release.cmake
 delete mode 100644 triplets/community/x64-windows-release.cmake

diff --git a/ports/ffmpeg/portfile.cmake b/ports/ffmpeg/portfile.cmake
index 35bac875b..96c9247f6 100644
--- a/ports/ffmpeg/portfile.cmake
+++ b/ports/ffmpeg/portfile.cmake
@@ -42,7 +42,7 @@ else()
     set(LIB_PATH_VAR "LIBRARY_PATH")
 endif()
 
-set(OPTIONS "--enable-pic --disable-doc --enable-debug --enable-runtime-cpudetect --disable-autodetect")
+set(OPTIONS "--enable-pic --disable-doc --enable-runtime-cpudetect --disable-autodetect")
 
 if(VCPKG_TARGET_IS_WINDOWS)
     vcpkg_acquire_msys(MSYS_ROOT PACKAGES automake1.16)
diff --git a/scripts/toolchains/windows.cmake b/scripts/toolchains/windows.cmake
index 7f4886815..57ebcae92 100644
--- a/scripts/toolchains/windows.cmake
+++ b/scripts/toolchains/windows.cmake
@@ -56,25 +56,25 @@ if(NOT _CMAKE_IN_TRY_COMPILE)
         set(MP_BUILD_FLAG "/MP")
     endif()
 
-    set(CMAKE_CXX_FLAGS " /nologo /DWIN32 /D_WINDOWS /W3 ${CHARSET_FLAG} /GR /EHsc ${MP_BUILD_FLAG} ${VCPKG_CXX_FLAGS}" CACHE STRING "")
+    set(CMAKE_CXX_FLAGS " /nologo /DWIN32 /D_WINDOWS /W3 ${CHARSET_FLAG} /GR /EHsc /Zc:__cplusplus ${MP_BUILD_FLAG} ${VCPKG_CXX_FLAGS}" CACHE STRING "")
     set(CMAKE_C_FLAGS " /nologo /DWIN32 /D_WINDOWS /W3 ${CHARSET_FLAG} ${MP_BUILD_FLAG} ${VCPKG_C_FLAGS}" CACHE STRING "")
 
     if(VCPKG_TARGET_ARCHITECTURE STREQUAL "arm64ec")
         string(APPEND CMAKE_CXX_FLAGS " /arm64EC /D_AMD64_ /DAMD64 /D_ARM64EC_ /DARM64EC")
         string(APPEND CMAKE_C_FLAGS " /arm64EC /D_AMD64_ /DAMD64 /D_ARM64EC_ /DARM64EC")
     endif()
-    set(CMAKE_RC_FLAGS "-c65001 /DWIN32" CACHE STRING "")
+    set(CMAKE_RC_FLAGS "-c65001 -DWIN32 -nologo" CACHE STRING "")
 
     unset(CHARSET_FLAG)
 
-    set(CMAKE_CXX_FLAGS_DEBUG "/D_DEBUG ${VCPKG_CRT_LINK_FLAG_PREFIX}d /Z7 /Ob0 /Od /RTC1 ${VCPKG_CXX_FLAGS_DEBUG}" CACHE STRING "")
-    set(CMAKE_C_FLAGS_DEBUG "/D_DEBUG ${VCPKG_CRT_LINK_FLAG_PREFIX}d /Z7 /Ob0 /Od /RTC1 ${VCPKG_C_FLAGS_DEBUG}" CACHE STRING "")
-    set(CMAKE_CXX_FLAGS_RELEASE "${VCPKG_CRT_LINK_FLAG_PREFIX} /O2 /Oi /Gy /DNDEBUG /Z7 ${VCPKG_CXX_FLAGS_RELEASE}" CACHE STRING "")
-    set(CMAKE_C_FLAGS_RELEASE "${VCPKG_CRT_LINK_FLAG_PREFIX} /O2 /Oi /Gy /DNDEBUG /Z7 ${VCPKG_C_FLAGS_RELEASE}" CACHE STRING "")
+    set(CMAKE_CXX_FLAGS_DEBUG "/D_DEBUG ${VCPKG_CRT_LINK_FLAG_PREFIX}d /Zi /Ob0 /Od /RTC1 ${VCPKG_CXX_FLAGS_DEBUG}" CACHE STRING "")
+    set(CMAKE_C_FLAGS_DEBUG "/D_DEBUG ${VCPKG_CRT_LINK_FLAG_PREFIX}d /Zi /Ob0 /Od /RTC1 ${VCPKG_C_FLAGS_DEBUG}" CACHE STRING "")
+    set(CMAKE_CXX_FLAGS_RELEASE "${VCPKG_CRT_LINK_FLAG_PREFIX} /O2 /Ob3 /Oi /Gy /Gw /DNDEBUG /fp:fast /QIntel-jcc-erratum /Zi /Zc:inline ${VCPKG_CXX_FLAGS_RELEASE}" CACHE STRING "")
+    set(CMAKE_C_FLAGS_RELEASE "${VCPKG_CRT_LINK_FLAG_PREFIX} /O2 /Ob3 /Oi /Gy /Gw /DNDEBUG /fp:fast /QIntel-jcc-erratum /Zi /Zc:inline ${VCPKG_C_FLAGS_RELEASE}" CACHE STRING "")
 
     string(APPEND CMAKE_STATIC_LINKER_FLAGS_RELEASE_INIT " /nologo ")
-    set(CMAKE_SHARED_LINKER_FLAGS_RELEASE "/nologo /DEBUG /INCREMENTAL:NO /OPT:REF /OPT:ICF ${VCPKG_LINKER_FLAGS} ${VCPKG_LINKER_FLAGS_RELEASE}" CACHE STRING "")
-    set(CMAKE_EXE_LINKER_FLAGS_RELEASE "/nologo /DEBUG /INCREMENTAL:NO /OPT:REF /OPT:ICF ${VCPKG_LINKER_FLAGS} ${VCPKG_LINKER_FLAGS_RELEASE}" CACHE STRING "")
+    set(CMAKE_SHARED_LINKER_FLAGS_RELEASE "/nologo /DEBUG /INCREMENTAL:NO /OPT:REF /OPT:ICF /OPT:LBR /DYNAMICBASE /FIXED:NO /NXCOMPAT /LARGEADDRESSAWARE ${VCPKG_LINKER_FLAGS} ${VCPKG_LINKER_FLAGS_RELEASE}" CACHE STRING "")
+    set(CMAKE_EXE_LINKER_FLAGS_RELEASE "/nologo /DEBUG /INCREMENTAL:NO /OPT:REF /OPT:ICF /OPT:LBR /DYNAMICBASE /FIXED:NO /NXCOMPAT /LARGEADDRESSAWARE ${VCPKG_LINKER_FLAGS} ${VCPKG_LINKER_FLAGS_RELEASE}" CACHE STRING "")
 
     string(APPEND CMAKE_STATIC_LINKER_FLAGS_DEBUG_INIT " /nologo ")
     string(APPEND CMAKE_SHARED_LINKER_FLAGS_DEBUG_INIT " /nologo ${VCPKG_LINKER_FLAGS} ${VCPKG_LINKER_FLAGS_DEBUG} ")
diff --git a/triplets/arm-uwp.cmake b/triplets/arm-uwp.cmake
index d632c2211..d233bb250 100644
--- a/triplets/arm-uwp.cmake
+++ b/triplets/arm-uwp.cmake
@@ -1,6 +1,7 @@
 set(VCPKG_TARGET_ARCHITECTURE arm)
 set(VCPKG_CRT_LINKAGE dynamic)
 set(VCPKG_LIBRARY_LINKAGE dynamic)
+set(VCPKG_BUILD_TYPE release)
 
 set(VCPKG_CMAKE_SYSTEM_NAME WindowsStore)
 set(VCPKG_CMAKE_SYSTEM_VERSION 10.0)
diff --git a/triplets/arm64-windows.cmake b/triplets/arm64-windows.cmake
index 8769e975d..1220465c9 100644
--- a/triplets/arm64-windows.cmake
+++ b/triplets/arm64-windows.cmake
@@ -1,3 +1,4 @@
 set(VCPKG_TARGET_ARCHITECTURE arm64)
 set(VCPKG_CRT_LINKAGE dynamic)
 set(VCPKG_LIBRARY_LINKAGE dynamic)
+set(VCPKG_BUILD_TYPE release)
diff --git a/triplets/community/arm-android.cmake b/triplets/community/arm-android.cmake
index 1376baede..af8e808a9 100644
--- a/triplets/community/arm-android.cmake
+++ b/triplets/community/arm-android.cmake
@@ -1,6 +1,8 @@
 set(VCPKG_TARGET_ARCHITECTURE arm)
 set(VCPKG_CRT_LINKAGE static)
 set(VCPKG_LIBRARY_LINKAGE static)
+set(VCPKG_BUILD_TYPE release)
+
 set(VCPKG_CMAKE_SYSTEM_NAME Android)
 set(VCPKG_MAKE_BUILD_TRIPLET "--host=armv7a-linux-androideabi")
 set(VCPKG_CMAKE_CONFIGURE_OPTIONS -DANDROID_ABI=armeabi-v7a -DANDROID_ARM_NEON=OFF)
diff --git a/triplets/community/arm-ios.cmake b/triplets/community/arm-ios.cmake
index 7fee3751d..e3b5fe327 100644
--- a/triplets/community/arm-ios.cmake
+++ b/triplets/community/arm-ios.cmake
@@ -1,4 +1,6 @@
 set(VCPKG_TARGET_ARCHITECTURE arm)
 set(VCPKG_CRT_LINKAGE dynamic)
 set(VCPKG_LIBRARY_LINKAGE static)
+set(VCPKG_BUILD_TYPE release)
+
 set(VCPKG_CMAKE_SYSTEM_NAME iOS)
diff --git a/triplets/community/arm-linux.cmake b/triplets/community/arm-linux.cmake
index 931d1ad5e..490455fb1 100644
--- a/triplets/community/arm-linux.cmake
+++ b/triplets/community/arm-linux.cmake
@@ -1,5 +1,6 @@
 set(VCPKG_TARGET_ARCHITECTURE arm)
 set(VCPKG_CRT_LINKAGE dynamic)
 set(VCPKG_LIBRARY_LINKAGE static)
+set(VCPKG_BUILD_TYPE release)
 
 set(VCPKG_CMAKE_SYSTEM_NAME Linux)
diff --git a/triplets/community/arm-mingw-dynamic.cmake b/triplets/community/arm-mingw-dynamic.cmake
index b23870c11..338b2256d 100644
--- a/triplets/community/arm-mingw-dynamic.cmake
+++ b/triplets/community/arm-mingw-dynamic.cmake
@@ -1,7 +1,8 @@
 set(VCPKG_TARGET_ARCHITECTURE arm)
 set(VCPKG_CRT_LINKAGE dynamic)
 set(VCPKG_LIBRARY_LINKAGE dynamic)
+set(VCPKG_BUILD_TYPE release)
 set(VCPKG_ENV_PASSTHROUGH PATH)
 
 set(VCPKG_CMAKE_SYSTEM_NAME MinGW)
-set(VCPKG_POLICY_DLLS_WITHOUT_LIBS enabled)
+set(VCPKG_POLICY_DLLS_WITHOUT_LIBS enabled)
diff --git a/triplets/community/arm-mingw-static.cmake b/triplets/community/arm-mingw-static.cmake
index ca99fac83..5673cf248 100644
--- a/triplets/community/arm-mingw-static.cmake
+++ b/triplets/community/arm-mingw-static.cmake
@@ -1,6 +1,7 @@
 set(VCPKG_TARGET_ARCHITECTURE arm)
 set(VCPKG_CRT_LINKAGE dynamic)
 set(VCPKG_LIBRARY_LINKAGE static)
+set(VCPKG_BUILD_TYPE release)
 set(VCPKG_ENV_PASSTHROUGH PATH)
 
 set(VCPKG_CMAKE_SYSTEM_NAME MinGW)
diff --git a/triplets/community/arm-neon-android.cmake b/triplets/community/arm-neon-android.cmake
index a26d9aa3e..1cc794011 100644
--- a/triplets/community/arm-neon-android.cmake
+++ b/triplets/community/arm-neon-android.cmake
@@ -1,6 +1,8 @@
 set(VCPKG_TARGET_ARCHITECTURE arm)
 set(VCPKG_CRT_LINKAGE static)
 set(VCPKG_LIBRARY_LINKAGE static)
+set(VCPKG_BUILD_TYPE release)
+
 set(VCPKG_CMAKE_SYSTEM_NAME Android)
 set(VCPKG_MAKE_BUILD_TRIPLET "--host=armv7a-linux-androideabi")
 set(VCPKG_CMAKE_CONFIGURE_OPTIONS -DANDROID_ABI=armeabi-v7a -DANDROID_ARM_NEON=ON)
diff --git a/triplets/community/arm-windows-static.cmake b/triplets/community/arm-windows-static.cmake
index 9ba5feed7..d38cc524f 100644
--- a/triplets/community/arm-windows-static.cmake
+++ b/triplets/community/arm-windows-static.cmake
@@ -1,3 +1,4 @@
 set(VCPKG_TARGET_ARCHITECTURE arm)
 set(VCPKG_CRT_LINKAGE static)
 set(VCPKG_LIBRARY_LINKAGE static)
+set(VCPKG_BUILD_TYPE release)
diff --git a/triplets/community/arm-windows.cmake b/triplets/community/arm-windows.cmake
index 3aa75933f..ebd6c9443 100644
--- a/triplets/community/arm-windows.cmake
+++ b/triplets/community/arm-windows.cmake
@@ -1,5 +1,6 @@
 set(VCPKG_TARGET_ARCHITECTURE arm)
 set(VCPKG_CRT_LINKAGE dynamic)
 set(VCPKG_LIBRARY_LINKAGE dynamic)
+set(VCPKG_BUILD_TYPE release)
 
 
diff --git a/triplets/community/arm64-android.cmake b/triplets/community/arm64-android.cmake
index 784f0d39c..4206fc229 100644
--- a/triplets/community/arm64-android.cmake
+++ b/triplets/community/arm64-android.cmake
@@ -1,6 +1,8 @@
 set(VCPKG_TARGET_ARCHITECTURE arm64)
 set(VCPKG_CRT_LINKAGE static)
 set(VCPKG_LIBRARY_LINKAGE static)
+set(VCPKG_BUILD_TYPE release)
+
 set(VCPKG_CMAKE_SYSTEM_NAME Android)
 set(VCPKG_MAKE_BUILD_TRIPLET "--host=aarch64-linux-android")
 set(VCPKG_CMAKE_CONFIGURE_OPTIONS -DANDROID_ABI=arm64-v8a)
diff --git a/triplets/community/arm64-ios.cmake b/triplets/community/arm64-ios.cmake
index c632e9381..e9800ae14 100644
--- a/triplets/community/arm64-ios.cmake
+++ b/triplets/community/arm64-ios.cmake
@@ -1,4 +1,6 @@
 set(VCPKG_TARGET_ARCHITECTURE arm64)
 set(VCPKG_CRT_LINKAGE dynamic)
 set(VCPKG_LIBRARY_LINKAGE static)
+set(VCPKG_BUILD_TYPE release)
+
 set(VCPKG_CMAKE_SYSTEM_NAME iOS)
diff --git a/triplets/community/arm64-linux.cmake b/triplets/community/arm64-linux.cmake
index 07452e099..2b1ef90a6 100644
--- a/triplets/community/arm64-linux.cmake
+++ b/triplets/community/arm64-linux.cmake
@@ -1,5 +1,6 @@
 set(VCPKG_TARGET_ARCHITECTURE arm64)
 set(VCPKG_CRT_LINKAGE dynamic)
 set(VCPKG_LIBRARY_LINKAGE static)
+set(VCPKG_BUILD_TYPE release)
 
 set(VCPKG_CMAKE_SYSTEM_NAME Linux)
diff --git a/triplets/community/arm64-mingw-dynamic.cmake b/triplets/community/arm64-mingw-dynamic.cmake
index 30ed204ed..e9b43ad5d 100644
--- a/triplets/community/arm64-mingw-dynamic.cmake
+++ b/triplets/community/arm64-mingw-dynamic.cmake
@@ -1,6 +1,7 @@
 set(VCPKG_TARGET_ARCHITECTURE arm64)
 set(VCPKG_CRT_LINKAGE dynamic)
 set(VCPKG_LIBRARY_LINKAGE dynamic)
+set(VCPKG_BUILD_TYPE release)
 set(VCPKG_ENV_PASSTHROUGH PATH)
 
 set(VCPKG_CMAKE_SYSTEM_NAME MinGW)
diff --git a/triplets/community/arm64-mingw-static.cmake b/triplets/community/arm64-mingw-static.cmake
index c4d88830f..7d1a9256d 100644
--- a/triplets/community/arm64-mingw-static.cmake
+++ b/triplets/community/arm64-mingw-static.cmake
@@ -1,6 +1,7 @@
 set(VCPKG_TARGET_ARCHITECTURE arm64)
 set(VCPKG_CRT_LINKAGE dynamic)
 set(VCPKG_LIBRARY_LINKAGE static)
+set(VCPKG_BUILD_TYPE release)
 set(VCPKG_ENV_PASSTHROUGH PATH)
 
 set(VCPKG_CMAKE_SYSTEM_NAME MinGW)
diff --git a/triplets/community/arm64-osx-dynamic.cmake b/triplets/community/arm64-osx-dynamic.cmake
index c9b1243c9..194f5c7c0 100644
--- a/triplets/community/arm64-osx-dynamic.cmake
+++ b/triplets/community/arm64-osx-dynamic.cmake
@@ -1,6 +1,7 @@
 set(VCPKG_TARGET_ARCHITECTURE arm64)
 set(VCPKG_CRT_LINKAGE dynamic)
 set(VCPKG_LIBRARY_LINKAGE dynamic)
+set(VCPKG_BUILD_TYPE release)
 
 set(VCPKG_CMAKE_SYSTEM_NAME Darwin)
 set(VCPKG_OSX_ARCHITECTURES arm64)
diff --git a/triplets/community/arm64-osx.cmake b/triplets/community/arm64-osx.cmake
index 62325a696..22c04c9f4 100644
--- a/triplets/community/arm64-osx.cmake
+++ b/triplets/community/arm64-osx.cmake
@@ -1,6 +1,7 @@
 set(VCPKG_TARGET_ARCHITECTURE arm64)
 set(VCPKG_CRT_LINKAGE dynamic)
 set(VCPKG_LIBRARY_LINKAGE static)
+set(VCPKG_BUILD_TYPE release)
 
 set(VCPKG_CMAKE_SYSTEM_NAME Darwin)
 set(VCPKG_OSX_ARCHITECTURES arm64)
diff --git a/triplets/community/arm64-uwp.cmake b/triplets/community/arm64-uwp.cmake
index b37c34e87..e9dbf6a52 100644
--- a/triplets/community/arm64-uwp.cmake
+++ b/triplets/community/arm64-uwp.cmake
@@ -1,6 +1,7 @@
 set(VCPKG_TARGET_ARCHITECTURE arm64)
 set(VCPKG_CRT_LINKAGE dynamic)
 set(VCPKG_LIBRARY_LINKAGE dynamic)
+set(VCPKG_BUILD_TYPE release)
 
 set(VCPKG_CMAKE_SYSTEM_NAME WindowsStore)
 set(VCPKG_CMAKE_SYSTEM_VERSION 10.0)
diff --git a/triplets/community/arm64-windows-static-md.cmake b/triplets/community/arm64-windows-static-md.cmake
index f798113e1..f2a83e5ef 100644
--- a/triplets/community/arm64-windows-static-md.cmake
+++ b/triplets/community/arm64-windows-static-md.cmake
@@ -1,3 +1,4 @@
 set(VCPKG_TARGET_ARCHITECTURE arm64)
 set(VCPKG_CRT_LINKAGE dynamic)
 set(VCPKG_LIBRARY_LINKAGE static)
+set(VCPKG_BUILD_TYPE release)
diff --git a/triplets/community/arm64-windows-static.cmake b/triplets/community/arm64-windows-static.cmake
index 34c53f0a2..e6eb810f2 100644
--- a/triplets/community/arm64-windows-static.cmake
+++ b/triplets/community/arm64-windows-static.cmake
@@ -1,4 +1,4 @@
 set(VCPKG_TARGET_ARCHITECTURE arm64)
 set(VCPKG_CRT_LINKAGE static)
 set(VCPKG_LIBRARY_LINKAGE static)
-
+set(VCPKG_BUILD_TYPE release)
diff --git a/triplets/community/armv6-android.cmake b/triplets/community/armv6-android.cmake
index 7e6015cc9..367dafeb5 100644
--- a/triplets/community/armv6-android.cmake
+++ b/triplets/community/armv6-android.cmake
@@ -1,5 +1,7 @@
 set(VCPKG_TARGET_ARCHITECTURE arm)
 set(VCPKG_CRT_LINKAGE static)
 set(VCPKG_LIBRARY_LINKAGE static)
+set(VCPKG_BUILD_TYPE release)
+
 set(VCPKG_CMAKE_SYSTEM_NAME Android)
 set(VCPKG_CMAKE_CONFIGURE_OPTIONS -DANDROID_ABI=armeabi -DANDROID_ARM_MODE=arm)
diff --git a/triplets/community/ppc64le-linux.cmake b/triplets/community/ppc64le-linux.cmake
index 70fa5b44c..a1745a361 100644
--- a/triplets/community/ppc64le-linux.cmake
+++ b/triplets/community/ppc64le-linux.cmake
@@ -1,5 +1,6 @@
 set(VCPKG_TARGET_ARCHITECTURE ppc64le)
 set(VCPKG_CRT_LINKAGE dynamic)
 set(VCPKG_LIBRARY_LINKAGE static)
+set(VCPKG_BUILD_TYPE release)
 
 set(VCPKG_CMAKE_SYSTEM_NAME Linux)
diff --git a/triplets/community/s390x-linux.cmake b/triplets/community/s390x-linux.cmake
index 0aaefda4e..1f5b44e38 100644
--- a/triplets/community/s390x-linux.cmake
+++ b/triplets/community/s390x-linux.cmake
@@ -1,5 +1,6 @@
 set(VCPKG_TARGET_ARCHITECTURE s390x)
 set(VCPKG_CRT_LINKAGE dynamic)
 set(VCPKG_LIBRARY_LINKAGE static)
+set(VCPKG_BUILD_TYPE release)
 
 set(VCPKG_CMAKE_SYSTEM_NAME Linux)
diff --git a/triplets/community/wasm32-emscripten.cmake b/triplets/community/wasm32-emscripten.cmake
index ece5d8a22..897c5284f 100644
--- a/triplets/community/wasm32-emscripten.cmake
+++ b/triplets/community/wasm32-emscripten.cmake
@@ -20,5 +20,7 @@ endif()
 set(VCPKG_TARGET_ARCHITECTURE wasm32)
 set(VCPKG_CRT_LINKAGE dynamic)
 set(VCPKG_LIBRARY_LINKAGE static)
+set(VCPKG_BUILD_TYPE release)
+
 set(VCPKG_CMAKE_SYSTEM_NAME Emscripten)
 set(VCPKG_CHAINLOAD_TOOLCHAIN_FILE "${EMSCRIPTEN_ROOT}/cmake/Modules/Platform/Emscripten.cmake")
diff --git a/triplets/community/x64-android.cmake b/triplets/community/x64-android.cmake
index 28404f4ed..7201190dd 100644
--- a/triplets/community/x64-android.cmake
+++ b/triplets/community/x64-android.cmake
@@ -1,6 +1,8 @@
 set(VCPKG_TARGET_ARCHITECTURE x64)
 set(VCPKG_CRT_LINKAGE static)
 set(VCPKG_LIBRARY_LINKAGE static)
+set(VCPKG_BUILD_TYPE release)
+
 set(VCPKG_CMAKE_SYSTEM_NAME Android)
 set(VCPKG_MAKE_BUILD_TRIPLET "--host=x86_64-linux-android")
 set(VCPKG_CMAKE_CONFIGURE_OPTIONS -DANDROID_ABI=x86_64)
diff --git a/triplets/community/x64-freebsd.cmake b/triplets/community/x64-freebsd.cmake
index 035c5e86a..812f424bf 100644
--- a/triplets/community/x64-freebsd.cmake
+++ b/triplets/community/x64-freebsd.cmake
@@ -1,6 +1,7 @@
 set(VCPKG_TARGET_ARCHITECTURE x64)
 set(VCPKG_CRT_LINKAGE dynamic)
 set(VCPKG_LIBRARY_LINKAGE static)
+set(VCPKG_BUILD_TYPE release)
 
 set(VCPKG_CMAKE_SYSTEM_NAME FreeBSD)
 
diff --git a/triplets/community/x64-ios.cmake b/triplets/community/x64-ios.cmake
index 0c93d6c68..5a94aa22a 100644
--- a/triplets/community/x64-ios.cmake
+++ b/triplets/community/x64-ios.cmake
@@ -1,4 +1,6 @@
 set(VCPKG_TARGET_ARCHITECTURE x64)
 set(VCPKG_CRT_LINKAGE dynamic)
 set(VCPKG_LIBRARY_LINKAGE static)
+set(VCPKG_BUILD_TYPE release)
+
 set(VCPKG_CMAKE_SYSTEM_NAME iOS)
diff --git a/triplets/community/x64-linux-dynamic.cmake b/triplets/community/x64-linux-dynamic.cmake
index abccb40b1..ef19f6d63 100644
--- a/triplets/community/x64-linux-dynamic.cmake
+++ b/triplets/community/x64-linux-dynamic.cmake
@@ -1,6 +1,7 @@
 set(VCPKG_TARGET_ARCHITECTURE x64)
 set(VCPKG_CRT_LINKAGE dynamic)
 set(VCPKG_LIBRARY_LINKAGE dynamic)
+set(VCPKG_BUILD_TYPE release)
 
 set(VCPKG_CMAKE_SYSTEM_NAME Linux)
 
diff --git a/triplets/community/x64-linux-release.cmake b/triplets/community/x64-linux-release.cmake
deleted file mode 100644
index d7f3e2b25..000000000
--- a/triplets/community/x64-linux-release.cmake
+++ /dev/null
@@ -1,6 +0,0 @@
-set(VCPKG_TARGET_ARCHITECTURE x64)
-set(VCPKG_CRT_LINKAGE dynamic)
-set(VCPKG_LIBRARY_LINKAGE static)
-
-set(VCPKG_CMAKE_SYSTEM_NAME Linux)
-set(VCPKG_BUILD_TYPE release)
diff --git a/triplets/community/x64-mingw-dynamic.cmake b/triplets/community/x64-mingw-dynamic.cmake
index 1f132cfe1..424add870 100644
--- a/triplets/community/x64-mingw-dynamic.cmake
+++ b/triplets/community/x64-mingw-dynamic.cmake
@@ -1,7 +1,8 @@
 set(VCPKG_TARGET_ARCHITECTURE x64)
 set(VCPKG_CRT_LINKAGE dynamic)
 set(VCPKG_LIBRARY_LINKAGE dynamic)
+set(VCPKG_BUILD_TYPE release)
 set(VCPKG_ENV_PASSTHROUGH PATH)
 
 set(VCPKG_CMAKE_SYSTEM_NAME MinGW)
-set(VCPKG_POLICY_DLLS_WITHOUT_LIBS enabled)
+set(VCPKG_POLICY_DLLS_WITHOUT_LIBS enabled)
diff --git a/triplets/community/x64-mingw-static.cmake b/triplets/community/x64-mingw-static.cmake
index 28600f0cc..ae8c180c2 100644
--- a/triplets/community/x64-mingw-static.cmake
+++ b/triplets/community/x64-mingw-static.cmake
@@ -1,6 +1,7 @@
 set(VCPKG_TARGET_ARCHITECTURE x64)
 set(VCPKG_CRT_LINKAGE dynamic)
 set(VCPKG_LIBRARY_LINKAGE static)
+set(VCPKG_BUILD_TYPE release)
 set(VCPKG_ENV_PASSTHROUGH PATH)
 
 set(VCPKG_CMAKE_SYSTEM_NAME MinGW)
diff --git a/triplets/community/x64-openbsd.cmake b/triplets/community/x64-openbsd.cmake
index 9aa220a74..d4b663551 100644
--- a/triplets/community/x64-openbsd.cmake
+++ b/triplets/community/x64-openbsd.cmake
@@ -4,5 +4,6 @@
 set(VCPKG_TARGET_ARCHITECTURE x64)
 set(VCPKG_CRT_LINKAGE dynamic)
 set(VCPKG_LIBRARY_LINKAGE static)
+set(VCPKG_BUILD_TYPE release)
 
 set(VCPKG_CMAKE_SYSTEM_NAME OpenBSD)
diff --git a/triplets/community/x64-osx-dynamic.cmake b/triplets/community/x64-osx-dynamic.cmake
index 21fc20578..53a22353c 100644
--- a/triplets/community/x64-osx-dynamic.cmake
+++ b/triplets/community/x64-osx-dynamic.cmake
@@ -1,6 +1,7 @@
 set(VCPKG_TARGET_ARCHITECTURE x64)
 set(VCPKG_CRT_LINKAGE dynamic)
 set(VCPKG_LIBRARY_LINKAGE dynamic)
+set(VCPKG_BUILD_TYPE release)
 
 set(VCPKG_CMAKE_SYSTEM_NAME Darwin)
 set(VCPKG_OSX_ARCHITECTURES x86_64)
diff --git a/triplets/community/x64-osx-release.cmake b/triplets/community/x64-osx-release.cmake
deleted file mode 100644
index 64a2d07f5..000000000
--- a/triplets/community/x64-osx-release.cmake
+++ /dev/null
@@ -1,7 +0,0 @@
-set(VCPKG_TARGET_ARCHITECTURE x64)
-set(VCPKG_CRT_LINKAGE dynamic)
-set(VCPKG_LIBRARY_LINKAGE static)
-
-set(VCPKG_CMAKE_SYSTEM_NAME Darwin)
-set(VCPKG_OSX_ARCHITECTURES x86_64)
-set(VCPKG_BUILD_TYPE release)
diff --git a/triplets/community/x64-windows-release.cmake b/triplets/community/x64-windows-release.cmake
deleted file mode 100644
index f6c40253a..000000000
--- a/triplets/community/x64-windows-release.cmake
+++ /dev/null
@@ -1,4 +0,0 @@
-set(VCPKG_TARGET_ARCHITECTURE x64)
-set(VCPKG_CRT_LINKAGE dynamic)
-set(VCPKG_LIBRARY_LINKAGE dynamic)
-set(VCPKG_BUILD_TYPE release)
diff --git a/triplets/community/x64-windows-static-md.cmake b/triplets/community/x64-windows-static-md.cmake
index d7acd46a4..ac2ed5540 100644
--- a/triplets/community/x64-windows-static-md.cmake
+++ b/triplets/community/x64-windows-static-md.cmake
@@ -1,3 +1,4 @@
 set(VCPKG_TARGET_ARCHITECTURE x64)
 set(VCPKG_CRT_LINKAGE dynamic)
 set(VCPKG_LIBRARY_LINKAGE static)
+set(VCPKG_BUILD_TYPE release)
diff --git a/triplets/community/x86-android.cmake b/triplets/community/x86-android.cmake
index ef668467a..a54db12f7 100644
--- a/triplets/community/x86-android.cmake
+++ b/triplets/community/x86-android.cmake
@@ -1,6 +1,8 @@
 set(VCPKG_TARGET_ARCHITECTURE x86)
 set(VCPKG_CRT_LINKAGE static)
 set(VCPKG_LIBRARY_LINKAGE static)
+set(VCPKG_BUILD_TYPE release)
+
 set(VCPKG_CMAKE_SYSTEM_NAME Android)
 set(VCPKG_MAKE_BUILD_TRIPLET "--host=i686-linux-android")
 set(VCPKG_CMAKE_CONFIGURE_OPTIONS -DANDROID_ABI=x86)
diff --git a/triplets/community/x86-freebsd.cmake b/triplets/community/x86-freebsd.cmake
index 62042c766..acb2b7185 100644
--- a/triplets/community/x86-freebsd.cmake
+++ b/triplets/community/x86-freebsd.cmake
@@ -1,6 +1,7 @@
 set(VCPKG_TARGET_ARCHITECTURE x86)
 set(VCPKG_CRT_LINKAGE dynamic)
 set(VCPKG_LIBRARY_LINKAGE static)
+set(VCPKG_BUILD_TYPE release)
 
 set(VCPKG_CMAKE_SYSTEM_NAME FreeBSD)
 
diff --git a/triplets/community/x86-ios.cmake b/triplets/community/x86-ios.cmake
index 143b373b9..a7bf55d02 100644
--- a/triplets/community/x86-ios.cmake
+++ b/triplets/community/x86-ios.cmake
@@ -1,4 +1,6 @@
 set(VCPKG_TARGET_ARCHITECTURE x86)
 set(VCPKG_CRT_LINKAGE dynamic)
 set(VCPKG_LIBRARY_LINKAGE static)
+set(VCPKG_BUILD_TYPE release)
+
 set(VCPKG_CMAKE_SYSTEM_NAME iOS)
diff --git a/triplets/community/x86-mingw-dynamic.cmake b/triplets/community/x86-mingw-dynamic.cmake
index 000b36d96..7569a8817 100644
--- a/triplets/community/x86-mingw-dynamic.cmake
+++ b/triplets/community/x86-mingw-dynamic.cmake
@@ -1,7 +1,8 @@
 set(VCPKG_TARGET_ARCHITECTURE x86)
 set(VCPKG_CRT_LINKAGE dynamic)
 set(VCPKG_LIBRARY_LINKAGE dynamic)
+set(VCPKG_BUILD_TYPE release)
 set(VCPKG_ENV_PASSTHROUGH PATH)
 
 set(VCPKG_CMAKE_SYSTEM_NAME MinGW)
-set(VCPKG_POLICY_DLLS_WITHOUT_LIBS enabled)
+set(VCPKG_POLICY_DLLS_WITHOUT_LIBS enabled)
diff --git a/triplets/community/x86-mingw-static.cmake b/triplets/community/x86-mingw-static.cmake
index 3769b85f0..40f07288f 100644
--- a/triplets/community/x86-mingw-static.cmake
+++ b/triplets/community/x86-mingw-static.cmake
@@ -1,6 +1,7 @@
 set(VCPKG_TARGET_ARCHITECTURE x86)
 set(VCPKG_CRT_LINKAGE dynamic)
 set(VCPKG_LIBRARY_LINKAGE static)
+set(VCPKG_BUILD_TYPE release)
 set(VCPKG_ENV_PASSTHROUGH PATH)
 
 set(VCPKG_CMAKE_SYSTEM_NAME MinGW)
diff --git a/triplets/community/x86-uwp.cmake b/triplets/community/x86-uwp.cmake
index 13b9d77c2..e42a6d6e7 100644
--- a/triplets/community/x86-uwp.cmake
+++ b/triplets/community/x86-uwp.cmake
@@ -1,6 +1,7 @@
 set(VCPKG_TARGET_ARCHITECTURE x86)
 set(VCPKG_CRT_LINKAGE dynamic)
 set(VCPKG_LIBRARY_LINKAGE dynamic)
+set(VCPKG_BUILD_TYPE release)
 
 set(VCPKG_CMAKE_SYSTEM_NAME WindowsStore)
 set(VCPKG_CMAKE_SYSTEM_VERSION 10.0)
diff --git a/triplets/community/x86-windows-static-md.cmake b/triplets/community/x86-windows-static-md.cmake
index 22abeaf2d..e5b7ec8ea 100644
--- a/triplets/community/x86-windows-static-md.cmake
+++ b/triplets/community/x86-windows-static-md.cmake
@@ -1,3 +1,4 @@
 set(VCPKG_TARGET_ARCHITECTURE x86)
 set(VCPKG_CRT_LINKAGE dynamic)
 set(VCPKG_LIBRARY_LINKAGE static)
+set(VCPKG_BUILD_TYPE release)
diff --git a/triplets/community/x86-windows-static.cmake b/triplets/community/x86-windows-static.cmake
index c855f0caa..b4c73deca 100644
--- a/triplets/community/x86-windows-static.cmake
+++ b/triplets/community/x86-windows-static.cmake
@@ -1,3 +1,4 @@
 set(VCPKG_TARGET_ARCHITECTURE x86)
 set(VCPKG_CRT_LINKAGE static)
-set(VCPKG_LIBRARY_LINKAGE static)
\ No newline at end of file
+set(VCPKG_LIBRARY_LINKAGE static)
+set(VCPKG_BUILD_TYPE release)
\ No newline at end of file
diff --git a/triplets/community/x86-windows-v120.cmake b/triplets/community/x86-windows-v120.cmake
index 9cc6c315e..6e13cbab9 100644
--- a/triplets/community/x86-windows-v120.cmake
+++ b/triplets/community/x86-windows-v120.cmake
@@ -1,5 +1,7 @@
 set(VCPKG_TARGET_ARCHITECTURE x86)
 set(VCPKG_CRT_LINKAGE dynamic)
 set(VCPKG_LIBRARY_LINKAGE dynamic)
+set(VCPKG_BUILD_TYPE release)
+
 set(VCPKG_PLATFORM_TOOLSET "v120")
 set(VCPKG_DEP_INFO_OVERRIDE_VARS "v120")
\ No newline at end of file
diff --git a/triplets/x64-linux.cmake b/triplets/x64-linux.cmake
index 882213456..ab30045ea 100644
--- a/triplets/x64-linux.cmake
+++ b/triplets/x64-linux.cmake
@@ -1,6 +1,7 @@
 set(VCPKG_TARGET_ARCHITECTURE x64)
 set(VCPKG_CRT_LINKAGE dynamic)
 set(VCPKG_LIBRARY_LINKAGE static)
+set(VCPKG_BUILD_TYPE release)
 
 set(VCPKG_CMAKE_SYSTEM_NAME Linux)
 
diff --git a/triplets/x64-osx.cmake b/triplets/x64-osx.cmake
index 3867590df..6953986d8 100644
--- a/triplets/x64-osx.cmake
+++ b/triplets/x64-osx.cmake
@@ -1,6 +1,7 @@
 set(VCPKG_TARGET_ARCHITECTURE x64)
 set(VCPKG_CRT_LINKAGE dynamic)
 set(VCPKG_LIBRARY_LINKAGE static)
+set(VCPKG_BUILD_TYPE release)
 
 set(VCPKG_CMAKE_SYSTEM_NAME Darwin)
 set(VCPKG_OSX_ARCHITECTURES x86_64)
diff --git a/triplets/x64-uwp.cmake b/triplets/x64-uwp.cmake
index 902d8c238..edace61f3 100644
--- a/triplets/x64-uwp.cmake
+++ b/triplets/x64-uwp.cmake
@@ -1,6 +1,7 @@
 set(VCPKG_TARGET_ARCHITECTURE x64)
 set(VCPKG_CRT_LINKAGE dynamic)
 set(VCPKG_LIBRARY_LINKAGE dynamic)
+set(VCPKG_BUILD_TYPE release)
 
 set(VCPKG_CMAKE_SYSTEM_NAME WindowsStore)
 set(VCPKG_CMAKE_SYSTEM_VERSION 10.0)
diff --git a/triplets/x64-windows-static.cmake b/triplets/x64-windows-static.cmake
index 1581d2b0f..42894919c 100644
--- a/triplets/x64-windows-static.cmake
+++ b/triplets/x64-windows-static.cmake
@@ -1,4 +1,4 @@
 set(VCPKG_TARGET_ARCHITECTURE x64)
 set(VCPKG_CRT_LINKAGE static)
 set(VCPKG_LIBRARY_LINKAGE static)
-
+set(VCPKG_BUILD_TYPE release)
diff --git a/triplets/x64-windows.cmake b/triplets/x64-windows.cmake
index d0be7297f..f6c40253a 100644
--- a/triplets/x64-windows.cmake
+++ b/triplets/x64-windows.cmake
@@ -1,4 +1,4 @@
 set(VCPKG_TARGET_ARCHITECTURE x64)
 set(VCPKG_CRT_LINKAGE dynamic)
 set(VCPKG_LIBRARY_LINKAGE dynamic)
-
+set(VCPKG_BUILD_TYPE release)
diff --git a/triplets/x86-windows.cmake b/triplets/x86-windows.cmake
index 93c43c3ba..0a277bdb7 100644
--- a/triplets/x86-windows.cmake
+++ b/triplets/x86-windows.cmake
@@ -1,4 +1,4 @@
 set(VCPKG_TARGET_ARCHITECTURE x86)
 set(VCPKG_CRT_LINKAGE dynamic)
 set(VCPKG_LIBRARY_LINKAGE dynamic)
-
+set(VCPKG_BUILD_TYPE release)
-- 
2.41.0.windows.1

